mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/factories/FlinkCatalogFactory.java

```diff
@@ -70,31 +82,65 @@ public Catalog createCatalog(Context context) {
     final String defaultDatabase = helper.getOptions().get(DEFAULT_DATABASE);
     String metastoreUrl = helper.getOptions().get(ArcticCatalogFactoryOptions.METASTORE_URL);
 
+    String amoroCatalogName =
+        ArcticThriftUrl.parse(metastoreUrl, THRIFT_TABLE_SERVICE_NAME).catalogName();
+    UnifiedCatalog unifiedCatalog =
+        UnifiedCatalogLoader.loadUnifiedCatalog(metastoreUrl, amoroCatalogName, Maps.newHashMap());
+    Configuration hadoopConf = unifiedCatalog.authenticationContext().getConfiguration();
+
+    AmsClient client = new PooledAmsClient(metastoreUrl);
+    TableFormat catalogTableFormat;
+    try {
+      CatalogMeta catalogMeta = client.getCatalog(amoroCatalogName);
+      catalogTableFormat =
+          TableFormat.valueOf(catalogMeta.getCatalogProperties().get(TABLE_FORMATS));
+    } catch (Exception e) {
+      throw new IllegalStateException("failed when load catalog " + amoroCatalogName, e);
+    }
+
     Map<TableFormat, AbstractCatalog> availableCatalogs = Maps.newHashMap();
     SUPPORTED_FORMATS.forEach(
         tableFormat -> {
           if (!availableCatalogs.containsKey(tableFormat)) {
-            availableCatalogs.put(tableFormat, createCatalog(context, tableFormat));
+            if (catalogTableFormat == TableFormat.ICEBERG
+                && (tableFormat.in(TableFormat.MIXED_HIVE, TableFormat.MIXED_ICEBERG))) {
+              context
+                  .getOptions()
+                  .put(FLINK_TABLE_FORMATS.key(), TableFormat.MIXED_HIVE.toString());
+            }
+            availableCatalogs.put(tableFormat, createCatalog(context, tableFormat, hadoopConf));
```

Comment:
I can't understand this part of the code

Reply:
The main reason is Mixed catalog couldn't load the iceberg table. [Here](https://github.com/NetEase/amoro/blob/a87a2a98ad8db10dc49404f6837d39437503d3bc/core/src/main/java/com/netease/arctic/catalog/CatalogLoader.java#L98) is the validator.